include/swift/SIL/SILValue.h

```diff
@@ -128,12 +128,19 @@ struct ValueOwnershipKind {
   bool isTrivialOr(ValueOwnershipKind Kind) const {
     return Value == Trivial || Value == Kind;
   }
+
+  /// Given that this is a ValueOwnershipKind of a parent type of \p Proj,
+  /// return either trivial if the Proj is a trivial subtype or the parent value
+  /// ownership kind otherwise.
```

Comment:
"Given that there is an aggregate value (like a struct or enum) with this ownership kind, and a subobject of type Proj is being projected from the aggregate, return Trivial if Proj has trivial type and the aggregate's ownership kind otherwise.